### ISSUE TITLE
bam2cns doesn't handle spaces in the reference path

### ISSUE THREAD

**Issue by ressy** [BioInf-Wuerzburg/proovread#116]
bam2cns bombs out if the path to the reference file contains whitespace (I know, not my idea of a good filename either) because of how `glob()` works here at [line 244](https://github.com/BioInf-Wuerzburg/proovread/blob/master/bin/bam2cns#L244):

    ...
    if($opt_ref_pat){
    	($opt_ref_file) = glob($opt_ref_pat);
    	$v->exit("Reference file not found ($opt_ref_pat)") unless $opt_ref_file;
    ...

It doesn't get caught there since there's some text in `$opt_ref_file`, but later in `lib/Fastq/Parser.pm` when it tries to read the file.

Maybe something like this would work instead?

    	($opt_ref_file) = glob("'${opt_ref_pat}'");

I can make that a pull request if it's helpful.  Thanks for proovread by the way!

**Comment by thackl** [comment]
Cool, thanks for reporting the issue. I'm guessing you have tried your solution - I'm surprised this is the only thing that breaks with spaces in paths. I have to admit, I never tested for that.

If it works, I'm happy to merge a pull request.

**Comment by ressy** [comment]
Well, it works for me, but I'm not much of a Perl person so here's some more detail to hopefully show my thinking makes sense.  (And I wouldn't have noticed that either, except I had a script download a set of reference sequences and name them automatically, and one of the names happened to have a space.)

Before making any changes to the code, it works so long as there are no spaces or if I double-escape them for both the shell and Perl.  For example these first two commands run OK (in the second one bash takes the path with a space as a single argument and then presumably Perl understands that backslash) but the third one fails when it sets the reference path to just "example".

    $ proovread/bin/bam2cns --bam example_reads.bam --ref example_reference.fasta --prefix example_output
    $ proovread/bin/bam2cns --bam example_reads.bam --ref "example\ reference.fasta" --prefix example_output
    $ proovread/bin/bam2cns --bam example_reads.bam --ref "example reference.fasta" --prefix example_output
    Fastq::Parser::new: example, No such file or directory at <path>/proovread/bin/../lib/Fastq/Parser.pm line 229, <SAM> line 3.
    Use of uninitialized value in numeric eq (==) at <path>/proovread/bin/../lib/Fastq/Parser.pm line 694, <SAM> line 3.
            (in cleanup) Can't use an undefined value as a symbol reference at <path>/proovread/bin/../lib/Fastq/Parser.pm line 246, <SAM> line 3.

Then, if I change that glob() line like I described, any of those variants work.

**Comment by ressy** [comment]
OK, pull request created.  Oh and I'm on Perl 5.22.1 in an Anaconda 3 environment on Ubuntu.

**Comment by ressy** [comment]
Whoops, looks like that change breaks bam2cns for Perl < 5.18.3 ([thanks Travis](https://travis-ci.org/BioInf-Wuerzburg/proovread/builds/326974677)).  I'll see if there's an easy fix for that, but if not I'll just cancel the PR for the time being.

**Comment by thackl** [comment]
Yeah, too bad. Thanks for making the effort, and sorry are now getting dragged into the fabulous world of Perl madness ;). Don't waste too much time on it, though. I can take a look at it next week as well.

**Comment by ressy** [comment]
Yeah, I just closed the pull request for now.  If I start digging into Perl more I'll get back to you!
